codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttMessageBuilders.java

```diff
@@ -328,6 +345,166 @@ public MqttConnAckMessage build() {
+            if (maximumPacketSize != null) {
+                props.add(new MqttProperties.IntegerProperty(MqttPropertyType.MAXIMUM_PACKET_SIZE.value(),
+                        maximumPacketSize.intValue()));
+            }
+            props.add(new MqttProperties.IntegerProperty(MqttPropertyType.TOPIC_ALIAS_MAXIMUM.value(),
+                    topicAliasMaximum));
+            if (reasonString != null) {
+                props.add(new MqttProperties.StringProperty(MqttPropertyType.REASON_STRING.value(), reasonString));
+            }
+            props.add(userProperties);
+            if (wildcardSubscriptionAvailable != null) {
+                props.add(new MqttProperties.IntegerProperty(MqttPropertyType.WILDCARD_SUBSCRIPTION_AVAILABLE.value(),
+                        wildcardSubscriptionAvailable ? 1 : 0));
+            }
+            if (subscriptionIdentifiersAvailable != null) {
+                props.add(new MqttProperties.IntegerProperty(MqttPropertyType.SUBSCRIPTION_IDENTIFIER_AVAILABLE.value(),
+                        subscriptionIdentifiersAvailable ? 1 : 0));
+            }
+            if (sharedSubscriptionAvailable != null) {
+                props.add(new MqttProperties.IntegerProperty(MqttPropertyType.SHARED_SUBSCRIPTION_AVAILABLE.value(),
+                        sharedSubscriptionAvailable ? 1 : 0));
+            }
+            if (serverKeepAlive != null) {
+                props.add(new MqttProperties.IntegerProperty(MqttPropertyType.SERVER_KEEP_ALIVE.value(),
+                        serverKeepAlive));
+            }
+            if (responseInformation != null) {
+                props.add(new MqttProperties.StringProperty(MqttPropertyType.RESPONSE_INFORMATION.value(),
+                        responseInformation));
+            }
+            if (serverReference != null) {
+                props.add(new MqttProperties.StringProperty(MqttPropertyType.SERVER_REFERENCE.value(),
+                        serverReference));
+            }
+            if (authenticationMethod != null) {
+                props.add(new MqttProperties.StringProperty(MqttPropertyType.AUTHENTICATION_METHOD.value(),
+                        authenticationMethod));
+            }
+            if (authenticationData != null) {
+                props.add(new MqttProperties.BinaryProperty(MqttPropertyType.AUTHENTICATION_DATA.value(),
+                        authenticationData));
+            }
+
+            return props;
+        }
+
+        public void sessionExpiryInterval(long seconds) {
+            this.sessionExpiryInterval = seconds;
+        }
+
+        public void receiveMaximum(int value) {
+            if (value <= 0) {
+                throw new IllegalArgumentException("receive maximum property must be > 0");
+            }
+            this.receiveMaximum = value;
+        }
+
+        public void maximumQos(byte value) {
```

Comment:
usually builders should return itself so you can use method chaining

Reply:
the intent was to have a declarative listing, something like:

```kotlin
properties.apply {
   maximumQos = 1
   assignedClientId = "client1234"
   ....
}
```

to mimic in Java as:
```java
withProperties(p -> {
    p.maximumQos(1);
    p.assignedClientId("client1234");
   ...
})
```

Reply:
Still I think we should return `this`. It's up to the user to use it or not.

Reply:
I'm convinced, I applied the change, thanks to point it out